master/buildbot/scripts/base.py

```diff
@@ -58,6 +58,7 @@ def isBuildBotRunning(basedir, quiet):
                 printMessage(
                         message="Removing twistd.pid, file has pid {} but buildbot is not running".format(pid),
                         quiet=quiet)
+                f.close()
```

Comment:
Oh, you're right. For some reason, I was thinking that that IF statement was still using pidfile, so it would take more lines of code to have the file deletion part outside of the With enclosure.